gramlib/grammar.ml

```diff
@@ -85,6 +92,17 @@ let tokens con =
     egram.gtokens;
   !list
 
+type ty_norec
```

Comment:
I think you should export a manifest type here, like e.g. ``type ty_norec = [ `ty_norec ]`` and similarly for `ty_mayrec`. Otherwise the type-checker cannot unify the two types, but doesn't know they are different either. This is not mandatory for this PR to get it, but I think it would be an enhancement API-wise.

Reply:
Indeed, fixed, thanks!